File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    removeDuplicates(nums) {
        // Method - 1 using Set
        // let unique = Array.from(new Set(nums)).sort((a,b) => a - b);
        // for(let i=0; i<unique.length; i++) {
        //     nums[i] = unique[i];
        // }

        // return unique.length;

        // Method - 2 using two pointer
        let l = 1;
        for(let r=0 ;r<nums.length; r++) {
            if(nums[r] != nums[r+1]) {
                nums[l] = nums[r+1];
                l++;
            }
        }

        return l-1;
        
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-1.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> ans;
        int n = nums.size();

        for(int i=0; i<2*n; i++){
            if(i >= n) {
                ans.push_back(nums[i-n]);
                continue;
            }
            ans.push_back(nums[i]);
        }

        // for(int i=n; i<2*n; i++) {
        //     ans.push_back(nums[i-n]);
        // }

        return ans;
    }
};